homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,229 @@
+"""Config flow for LastFm."""
+from __future__ import annotations
+
+from typing import Any
+
+from pylast import LastFMNetwork, User, WSError
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    OptionsFlowWithConfigEntry,
+)
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_MAIN_USER, CONF_USERS, DOMAIN
+
+PLACEHOLDERS = {"api_account_url": "https://www.last.fm/api/account/create"}
+
+CONFIG_SCHEMA: vol.Schema = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Required(CONF_MAIN_USER): str,
+    }
+)
+
+
+def get_lastfm_user(api_key: str, username: str) -> User:
+    """Get lastFM User."""
+    return LastFMNetwork(api_key=api_key).get_user(username)
```

Comment:
What happens if the api key is wrong?
If it doesn't raise but it wouldn't return anything what do we do then in the flow?

Reply:
Perhaps it would be better to put `get_lastfm_user` and `validate_lastfm_user` in same function as validation for the main user?